### PR TITLE
Ordering of different types

### PR DESCRIPTION
**Pull request by hedengran**
Adds details on how ordering is defined between different types.

### CODE REVIEW

File: cypher/cypher-docs/src/docs/dev/syntax/comparison.asciidoc [hunk 1]
```diff
@@ -63,7 +63,26 @@ The following points give some details on how the comparison is performed.
  A leap year occurs on a year with an ordinal number divisible by `4`, that is not divisible by `100`, unless it divisible by `400`.
  This means that over `400` years there are `((365 * 4 + 1) * 25 - 1) * 4 + 1 = 146097` days, which means an average of `365.2425` days per year.].
 * Comparing for ordering when one argument is `null` (e.g. `null < 3` is `null`).
-
+* *Ordering* of values with *different* types:
+ ** The ordering is, in ascending order, defined according to the following list:
+  *** `MAP` types
+   **** <<cypher-literal-maps, Regular maps>>
+   **** <<cypher-map-projection, `NODE` maps>>
+   **** <<cypher-map-projection, `RELATIONSHIP` maps>>
```
Comment (sherfert) [inline cypher/cypher-docs/src/docs/dev/syntax/comparison.asciidoc]: A discussion in CLG concluded some time ago that we want to move away from nodes and relationships being seen as `Maps`. This is also only present in opencypher documents, but not in our [type documentation](https://neo4j.com/docs/cypher-manual/current/syntax/values/)

I suggest we make this a flat instead:
```suggestion
  *** <<cypher-literal-maps, Maps>>
  *** <<???, Nodes>>
  *** <<???, Relationships>>
```

The link to `cypher-map-projection` seems unrelated, I suggest linking here instead for nodes and relationships: https://neo4j.com/docs/cypher-manual/current/syntax/values/#structural-types

File: cypher/cypher-docs/src/docs/dev/syntax/comparison.asciidoc [hunk 2]
```diff
@@ -63,7 +63,26 @@ The following points give some details on how the comparison is performed.
...
+  *** <<cypher-temporal-introduction, `LOCALDATETIME`>>
+  *** <<cypher-temporal-introduction, `DATE`>>
+  *** <<cypher-temporal-introduction, `TIME`>>
+  *** <<cypher-temporal-introduction, `LOCALTIME`>>
+  *** <<cypher-temporal-introduction, `DURATION`>>
+  *** <<cypher-expressions-general, `STRING`>>
+  *** <<cypher-expressions-general, `BOOLEAN`>>
+  *** <<cypher-expressions-general, `NUMBER`>>
+ ** The value `NULL` is considered larger than any value.
+ ** For the container types, elements of the containers use orderability to determine the order between them.
```
Comment (sherfert) [inline cypher/cypher-docs/src/docs/dev/syntax/comparison.asciidoc]: I don't think "orderability" is a phrase introduced anywhere in our documentation. Consider phrasing it more like "the elements of the container type are compared pairwise for ordering and thus determine the ordering of two container types ..."

I think this is orthogonal to the change with `*Ordering* of values with *different* types`. Ordering of container types could be its own bullet point with sub-bullet-points for lists and maps. We also seem to call them [Composite](https://neo4j.com/docs/cypher-manual/current/syntax/values/#composite-types) types and not container types.

File: cypher/cypher-docs/src/docs/dev/syntax/comparison.asciidoc [hunk 3]
```diff
@@ -63,7 +63,26 @@ The following points give some details on how the comparison is performed.
...
  This means that over `400` years there are `((365 * 4 + 1) * 25 - 1) * 4 + 1 = 146097` days, which means an average of `365.2425` days per year.].
 * Comparing for ordering when one argument is `null` (e.g. `null < 3` is `null`).
-
+* *Ordering* of values with *different* types:
+ ** The ordering is, in ascending order, defined according to the following list:
+  *** `MAP` types
+   **** <<cypher-literal-maps, Regular maps>>
+   **** <<cypher-map-projection, `NODE` maps>>
+   **** <<cypher-map-projection, `RELATIONSHIP` maps>>
+  *** <<cypher-lists, `LIST`>>
```
Comment (sherfert) [inline cypher/cypher-docs/src/docs/dev/syntax/comparison.asciidoc]: We don't seem to use UPPERCASE for our types in any other place, But rather CamelCase, e.g. here: https://neo4j.com/docs/cypher-manual/current/syntax/temporal/#cypher-temporal-introduction

**Review by sherfert — CHANGES_REQUESTED** [review]

**Review by sherfert — APPROVED** [review]

**Comment by davidoliverSP2** [comment]
We suspect that the builds for this PR will go green when #1015 is merged. I'll review this once we have completed that PR.

**Review by davidoliverSP2 — APPROVED** [review]
Thanks @hedengran

**Pull request merged**